ambari-metrics/ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/PhoenixHBaseAccessor.java

```diff
@@ -1052,12 +1058,21 @@ private void appendMetricFromResultSet(TimelineMetrics metrics, Condition condit
       } else {
         // No aggregation requested
         // Execution never goes here, function always contain at least 1 element
-        getTimelineMetricsFromResultSet(metrics, null, condition, rs);
+        getTimelineMetricsFromResultSet(metrics, null, condition, rs, isEventDownsampledMetric(metricName));
+      }
+    }
+  }
+
+  private boolean isEventDownsampledMetric(String metricName) {
+    for (String pattern : eventMetricPatterns) {
```

Comment:
It cannot be a simple metric name lookup since the number of metrics that could be 'event' type can be unbounded, and holding a Set of metric names may become costly.